extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultSystemBackendEngine.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.vault;
+
+import io.quarkus.vault.runtime.config.VaultRuntimeConfig;
+import io.quarkus.vault.sys.VaultHealth;
+import io.quarkus.vault.sys.VaultHealthStatus;
+import io.quarkus.vault.sys.VaultInit;
+import io.quarkus.vault.sys.VaultSealStatus;
+
+/**
+ * This service provides access to the system backend.
+ *
+ * @see VaultRuntimeConfig
+ */
+public interface VaultSystemBackendEngine {
+
+    /**
+     * Initializes a new Vault.
+     * 
+     * @param secretShares specifies the number of shares to split the master key into.
+     * @param secretThreshold specifies the number of shares required to reconstruct the master key.
+     * @return Vault Init.
+     */
+    VaultInit init(int secretShares, int secretThreshold);
```

Comment:
We don't have `init` methods exposed anywhere else in the public Vault interfaces, this is an admin operation which should not be visible to the JAX-RS service code having `VaultSystemBackendEngine` injected

Reply:
We agreed with @vsevel to move the sys operations here.

Reply:
@lordofthejars OK, @vsevel Hi Vincent, do you feel it would be useful to have some users doing the Vault Initialization in scope of the current call ?

Reply:
@vsevel I'm asking because we've tried hard to have all Vault interfaces very user centric in a sense that each of those operations can be do something meaningful in scope of a current HTTP request, and `init` does not fit. However if you do feel it is worth it then I'll be good with it, thanks

Reply:
when I talked about moving some methods,I had specifically the `sys/health` in mind, but did not make it clear. I know @sberyozkin we had long discussion about this :) I think there is value in having those operations too on the `SystemBackendEngine`. and sometimes it confuses people to not find operations that we categorize as `admin` (although it is a valid criteria when priorizing what we want to work on first).
to make it short, now that @lordofthejars has moved all of them, I would not move them back.